nixos/modules/security/apparmor.nix

```diff
@@ -29,6 +39,577 @@ in
+      /usr/bin/refer		2 6 4
+      /usr/bin/soelim		2 6 4
+      /usr/bin/disable-paste	2 6 6
+      /usr/bin/troff		2 6 4
+      /usr/bin/strace-graph	2 6 4
+      /usr/bin/gpm-root	2 6 7
+      /usr/bin/hltest		2 6 7
+      /usr/bin/mev		2 6 6
+      /usr/bin/mouse-test	2 6 6
+      /usr/bin/strace		2 8 9
+      /usr/bin/scsiformat	2 7 10
+      /usr/bin/lsscsi		2 7 7
+      /usr/bin/scsiinfo	2 7 7
+      /usr/bin/sg_*		2 7 7
+      /usr/bin/build-classpath		2 6 6
+      /usr/bin/build-classpath-directory	2 6 6
+      /usr/bin/build-jar-repository		2 6 6
+      /usr/bin/diff-jars			2 6 6
+      /usr/bin/jvmjar				2 6 6
+      /usr/bin/rebuild-jar-repository		2 6 6
+      /usr/bin/scriptreplay	2 6 5
+      /usr/bin/cal		2 6 3
+      /usr/bin/chkdupexe	2 6 5
+      /usr/bin/col		2 6 4
+      /usr/bin/colcrt		2 6 4
+      /usr/bin/colrm		2 6 3
+      /usr/bin/column		2 6 4
+      /usr/bin/cytune		2 6 6
+      /usr/bin/ddate		2 6 3
+      /usr/bin/fdformat	2 6 6
+      /usr/bin/getopt		2 8 6
+      /usr/bin/hexdump	2 6 4
+      /usr/bin/hostid		2 6 4
+      /usr/bin/ipcrm		2 7 7
+      /usr/bin/ipcs		2 7 6
+      /usr/bin/isosize	2 6 4
+      /usr/bin/line		2 6 4
+      /usr/bin/look		2 6 5
+      /usr/bin/mcookie	2 7 5
+      /usr/bin/mesg		2 6 4
+      /usr/bin/namei		2 6 5
+      /usr/bin/rename		2 6 5
+      /usr/bin/renice		2 6 7
+      /usr/bin/rev		2 6 5
+      /usr/bin/script		2 6 6
+      /usr/bin/ChangeSymlinks	2 8 8
+      /usr/bin/setfdprm	2 6 7
+      /usr/bin/setsid		2 6 3
+      /usr/bin/setterm	2 6 5
+      /usr/bin/tailf		2 6 4
+      /usr/bin/time		2 6 4
+      /usr/bin/ul		2 6 4
+      /usr/bin/wall		2 6 5
+      /usr/bin/whereis	2 6 4
+      /usr/bin/which		2 6 3
+      /usr/bin/c_rehash	2 7 6
+      /usr/bin/openssl	2 8 6
+      /usr/bin/lsdev		2 6 5
+      /usr/bin/procinfo	2 6 5
+      /usr/bin/socklist	2 6 5
+      /usr/bin/filesize	2 6 3
+      /usr/bin/linkto		2 6 3
+      /usr/bin/mkinfodir	2 6 5
+      /usr/bin/old		2 6 4
+      /usr/bin/rpmlocate	2 6 5
+      /usr/bin/safe-rm	2 8 6
+      /usr/bin/safe-rmdir	2 8 6
+      /usr/bin/setJava	2 6 1
+      /usr/bin/vmstat		2 6 4
+      /usr/bin/top		2 6 6
+      /usr/bin/pinentry*	2 7 6
+      /usr/bin/free		2 8 4
+      /usr/bin/pmap		2 6 5
+      /usr/bin/slabtop	2 6 4
+      /usr/bin/tload		2 6 4
+      /usr/bin/watch		2 6 3
+      /usr/bin/w		2 6 4
+      /usr/bin/pstree.x11	2 6 4
+      /usr/bin/pstree		2 6 4
+      /usr/bin/snice		2 6 6
+      /usr/bin/skill		2 6 7
+      /usr/bin/pgrep		2 6 4
+      /usr/bin/killall	2 6 7
+      /usr/bin/curl		2 7 7
+      /usr/bin/slptool	2 7 8
+      /usr/bin/ldap*		2 7 7
+      /usr/bin/whatis		2 7 5
+     '';
+
+     # Enable logging to /var/log/messages so that
+     # aa-logprof can parse the text logfiles
+     services.journald.forwardToSyslog = true;
+     services.rsyslogd = {
+       enable = true;
+       extraConfig = ''
+         $ModLoad imklog
+         kern.*			     -/var/log/messages
+       '';
+     };
```

Comment:
Wouldn't it be preferable to use the Audit framework? 

And what if audit(d) is already enabled, and the user explicitly enables Apparmor? Won't this change break existing setups?